app/controllers/sessions_controller.rb

```diff
@@ -8,9 +8,16 @@ def new
   end
 
   def create
-    user = authenticate!(:password)
-    set_website(user.default_website)
-    after_login_redirect
+    user = User.find_by_email(user_params[:email])
+    unless user.nil?
+      if user.confirmation_token.nil?
+        user = authenticate!(:password)
+        set_website(user.default_website)
+        after_login_redirect
+      else
+        redirect_to login_url, alert: "Confirm your email first"
+      end
```

Comment:
all this code should be in warden. This way you can do a very simple check in there to see if it's confirmed or not and send a different message.